### PR TITLE
Don't ignore CONNECT packets with data that we don't know

### PR DESCRIPTION
**Pull request by heinrich5991**
This specifically affects 0.6.5. Just treat them the same way as those
without any data.

### CODE REVIEW

File: src/engine/shared/network_server.cpp [hunk 1]
```diff
@@ -570,6 +563,29 @@ int CNetServer::GetClientSlot(const NETADDR &Addr)
 	return Slot;
 }
 
+static bool IsDDNetControlMsg(const CNetPacketConstruct *pPacket)
+{
+	if(!(pPacket->m_Flags&NET_PACKETFLAG_CONTROL)
+		|| pPacket->m_DataSize < 1)
```
Comment (Learath2) [inline src/engine/shared/network_server.cpp]: This changes previous behaviour if `pPacket->m_DataSize == 1` is that intended?

File: src/engine/shared/network_server.cpp [hunk 2]
```diff
@@ -508,45 +508,38 @@ void CNetServer::OnConnCtrlMsg(NETADDR &Addr, int ClientID, int ControlMsg, cons
...
-
-		if (SupportsToken)
-		{
-			// response connection request with token
-			SECURITY_TOKEN Token = GetToken(Addr);
-			SendControl(Addr, NET_CTRLMSG_CONNECTACCEPT, SECURITY_TOKEN_MAGIC, sizeof(SECURITY_TOKEN_MAGIC), Token);
-		}
+		// response connection request with token
+		SECURITY_TOKEN Token = GetToken(Addr);
+		SendControl(Addr, NET_CTRLMSG_CONNECTACCEPT, SECURITY_TOKEN_MAGIC, sizeof(SECURITY_TOKEN_MAGIC), Token);
```
Comment (Learath2) [inline src/engine/shared/network_server.cpp]: This accepts clients that don't send a token as if they sent one. Is this intended? Why is this safe to do?

File: src/engine/shared/network_server.cpp [hunk 3]
```diff
@@ -643,9 +659,8 @@ int CNetServer::Recv(CNetChunk *pChunk)
 				{
 					// not found, client that wants to connect
 
-					if(m_RecvUnpacker.m_Data.m_Flags&NET_PACKETFLAG_CONTROL &&
-						m_RecvUnpacker.m_Data.m_DataSize > 1)
-						// got control msg with extra size (should support token)
+					if(IsDDNetControlMsg(&m_RecvUnpacker.m_Data))
```
Comment (Learath2) [inline src/engine/shared/network_server.cpp]: Seems this still ignores packets with data we don't know. Sounds contradictory with the commit message

Comment (heinrich5991) [reply src/engine/shared/network_server.cpp]: No, the ignoring previously happened in the `OnTokenCtrlMsg` function. It's now filtered before even calling that function and redirected to `OnPreConnMsg` below.

File: src/engine/shared/network_server.cpp (hunk 1 above)
Comment (heinrich5991) [reply src/engine/shared/network_server.cpp]: It doesn't really change behavior: The function should only be called when one of the two conditions below hold, and then the packet size is bigger than `1 + [4 +] sizeof(int)` anyway. This just guards against reading the `pPacket->m_aChunkData` if it's not big enough.

**Pull request merged**